web-client/src/modules/create/components/NewAddressModal.tsx

```diff
@@ -0,0 +1,158 @@
+  );
+  const userId = useSelector((state: AppState) => state.auth.user?.uid);
+
+  const handleFinish = () => {
+    geocode({
+      name: form.getFieldValue('name'),
+      address1: form.getFieldValue('address1'),
+      address2: form.getFieldValue('address2'),
+      city: form.getFieldValue('city'),
+      state: form.getFieldValue('state'),
+      postalCode: form.getFieldValue('postalCode'),
+      country: form.getFieldValue('country'),
+    })
+      .then(geocodedResult => {
+        modalSuccess(geocodedResult);
+        if (shouldSave) {
+          if (privilegedInformation && userId) {
+            const addressesToSave = {
+              ...addresses,
+            };
+            addressesToSave[geocodedResult.name] = geocodedResult;
+            privilegedInformation.addresses = addressesToSave;
+            dispatch(
+              updateUserPrivilegedInformation(userId, privilegedInformation),
+            );
+          }
+        }
+      })
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.error('could not geocode', error);
+      });
+  };
+
+  const { t } = useTranslation();
+  return (
+    <Modal
+      title="Basic Modal"
+      visible={visible}
+      onOk={handleFinish}
+      onCancel={closeModal}
+    >
+      <Form form={form} onFinish={handleFinish}>
+        <>
+          <Row gutter={[16, 6]}>
+            <Col span={24}>
+              <Form.Item
+                name="address1"
+                label={t('settings.changeAddressForm.address')}
+              >
+                <Input placeholder={t('address1')} />
+              </Form.Item>
+            </Col>
+          </Row>
+          <Row gutter={[16, 6]}>
+            <Col span={24}>
+              <Form.Item name="address2">
+                <Input placeholder={t('address2')} />
+              </Form.Item>
+            </Col>
+          </Row>
+          <Row>
+            <Col span={11}>
+              <Form.Item name="city">
+                <Input placeholder={t('city')} />
+              </Form.Item>
+            </Col>
+            <Col span={3} offset={1}>
+              <Form.Item name="state">
+                <Input placeholder={t('State')} />
+              </Form.Item>
+            </Col>
+            <Col span={5}>
+              <Form.Item name="postalCode">
+                <Input placeholder={t('code')} />
+              </Form.Item>
+            </Col>
+            <Col span={3} offset={1}>
+              <Form.Item name="country">
+                <Input placeholder={t('country')} />
+              </Form.Item>
+            </Col>
+          </Row>
+          {shouldSave && (
+            <Row gutter={[16, 6]}>
+              <Col span={24}>
+                <Form.Item
+                  name="name"
+                  required
+                  label={t('settings.changeAddressForm.nameOfAddress')}
+                >
+                  <Input id="nameofAddress" />
```

Comment:
Can you validate that this does not conflict with an existing address name?

Reply:
just added some code in handleFinish()